js/ui/overlay/ui.overlay.js

```diff
@@ -1103,6 +1106,49 @@ var Overlay = Widget.inherit({
         });
     },
 
+    _validateWheel(e) {
+        if(this._isWheelEvent(e)) {
+            var scrollableContainer = this._tryGetScrollableContainer(e.target);
+
+            if(scrollableContainer) {
+                if(this._isScrolledInMaxDirection(e, scrollableContainer)) {
+                    e.preventDefault();
+                }
+            }
+        }
+    },
+
+    _isWheelEvent(e) {
+        return e.type === "dxmousewheel";
+    },
+
+    _isScrolledInMaxDirection(e, container) {
+        var isReachedMax;
+        var isReachedMin;
+
+        if(e.delta > 0) {
+            isReachedMax = e.shiftKey ? !container.scrollLeft : !container.scrollTop;
+        } else {
+            if(e.shiftKey) {
+                isReachedMin = (container.clientWidth + container.scrollLeft) >= container.scrollWidth;
+            } else {
+                isReachedMin = (container.clientHeight + container.scrollTop) >= container.scrollHeight;
+            }
+        }
+
+        return (isReachedMax || isReachedMin);
+    },
+
+    _tryGetScrollableContainer(element) {
+        const scrollableClass = `.dx-scrollable-container`;
+
+        if($(element).is(scrollableClass)) {
```

Comment:
I think that event-based interaction between widgets will help to implement a more elegant solution.